### PR TITLE
OKAPI-911: Unit test failures for ProcessModuleHandleTest

### PR DESCRIPTION
**Pull request by adamdickmeiss**

### CODE REVIEW

File: okapi-core/src/main/java/org/folio/okapi/service/impl/ProcessModuleHandle.java [hunk 1]
```diff
@@ -125,48 +129,39 @@ private NuProcess launch(Vertx vertx, String id, EnvEntry[] env,
...
-              return;
-            }
-            c = cmdlineStart.replace("%p", Integer.toString(port));
-            l = new String[]{"sh", "-c", c};
-          } else {
-            future.fail("Can not deploy: No exec, no CmdlineStart in LaunchDescriptor");
-            return;
-          }
-          process = launch(vertx, id, env, l);
-          process.waitFor(1, TimeUnit.SECONDS);
```
Comment (julianladisch) [inline okapi-core/src/main/java/org/folio/okapi/service/impl/ProcessModuleHandle.java:154]: Can you also remove the import of TimeUnit?

File: okapi-core/src/main/java/org/folio/okapi/service/impl/ProcessModuleHandle.java [hunk 2]
```diff
@@ -72,6 +73,9 @@ public ProcessModuleHandle(Vertx vertx, LaunchDescriptor desc, String id,
 
   @Override
   public Future<Void> start() {
+    if (process != null) {
+      return Future.failedFuture("already started " + commandLine);
+    }
     if (port == 0) {
       return start2();
     }
```
Comment (julianladisch) [inline okapi-core/src/main/java/org/folio/okapi/service/impl/ProcessModuleHandle.java:81]: A few lines below we have this code:

      socket.close();
      return Future.failedFuture(messages.getMessage("11502", Integer.toString(port)));

This doesn't wait until the socket has been closed. Better:

      return socket.close().otherwiseEmpty()
          .compose(x -> Future.failedFuture(messages.getMessage("11502", Integer.toString(port))));

**Review by craigmcnally — APPROVED** [review]

File: okapi-core/src/main/java/org/folio/okapi/service/impl/ProcessModuleHandle.java [hunk 3]
```diff
@@ -125,48 +129,39 @@ private NuProcess launch(Vertx vertx, String id, EnvEntry[] env,
...
+      commandLine = cmdlineStart.replace("%p", Integer.toString(port));
+      l = new String[]{"sh", "-c", commandLine};
+    } else {
+      return Future.failedFuture("Can not deploy: No exec, no CmdlineStart in LaunchDescriptor");
+    }
+    final String commandLineF = commandLine;
+    process = launch(vertx, id, env, l);
+    Promise<Void> promise = Promise.promise();
+    vertx.setTimer(1000, timerRes -> {
+      if (!process.isRunning() && exitCode != 0) {
```
Comment (julianladisch) [inline okapi-core/src/main/java/org/folio/okapi/service/impl/ProcessModuleHandle.java:154]: Easier to read:

    if (process.isRunning() || exitCode == 0) {
      promise.complete();
      return;
    }
    if (exitCode == Integer.MIN_VALUE) {
      promise.fail(messages.getMessage("11504", commandLineF));
    } else {
      promise.fail(messages.getMessage("11500", exitCode));
    }

File: okapi-core/src/main/java/org/folio/okapi/service/impl/ProcessModuleHandle.java (hunk 2 above)
Comment (adamdickmeiss) [reply okapi-core/src/main/java/org/folio/okapi/service/impl/ProcessModuleHandle.java:80]: ok.

File: okapi-core/src/main/java/org/folio/okapi/service/impl/ProcessModuleHandle.java (hunk 3 above)
Comment (adamdickmeiss) [reply okapi-core/src/main/java/org/folio/okapi/service/impl/ProcessModuleHandle.java]: ok. In two places.

**Review by julianladisch — APPROVED** [review]

**Review by julianladisch — APPROVED** [review]

**Pull request merged**